dspace-spring-rest/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java

```diff
@@ -308,7 +307,112 @@ public void findOneTestWrongUUID() throws Exception {
+        Item publicItem = ItemBuilder.createItem(context, col1)
+                                      .withTitle("Public item 1")
+                                      .withIssueDate("2017-10-17")
+                                      .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                                      .withSubject("ExtraEntry")
+                                      .build();
+
+        //Add a bitstream to an item
+        String bitstreamContent = "ThisIsSomeDummyText";
+        Bitstream bitstream = null;
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+            bitstream = BitstreamBuilder.
+                                             createBitstream(context, publicItem, is)
+                                         .withName("Bitstream1")
+                                         .withMimeType("text/plain")
+                                         .build();
+        }
+
+        // Check publicItem creation
+        getClient().perform(get("/api/core/items/" + publicItem.getID()))
+                   .andExpect(status().isOk());
+
+        // Check publicItem bitstream creatino
+        getClient().perform(get("/api/core/items/" + publicItem.getID() + "/bitstreams"))
+                   .andExpect(status().isOk())
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$._links.self.href", Matchers
+                   .containsString("/api/core/items/" + publicItem.getID() + "/bitstreams")));
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        //Delete public item
+        getClient(token).perform(delete("/api/core/items/" + publicItem.getID()))
+                    .andExpect(status().is(204));
+
+        //Trying to get deleted item should fail with 404
+        getClient().perform(get("/api/core/items/" + publicItem.getID()))
+                   .andExpect(status().is(404));
+
+        //Trying to get deleted item bitstream should fail with 404
+        getClient().perform(get("/api/core/biststreams/" + bitstream.getID()))
+                   .andExpect(status().is(404));
+    }
+
+    public void deleteOneTemplateTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        //** GIVEN **
+        //1. A community with one collection.
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+
+        //2. One template item.
+        Item templateItem = ItemBuilder.createTemplateItem(context, col1);
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        //Check templateItem creation
+        getClient().perform(get("/api/core/items/" + templateItem.getID()))
```

Comment:
it is more useful to test that the item is still here after a failed attempt to delete it